src/main/java/io/opentracing/contrib/cassandra/TracingSession.java

```diff
@@ -219,31 +220,56 @@ public ResultSetFuture executeAsync(Statement statement) {
    */
   @Override
   public PreparedStatement prepare(String query) {
-    return session.prepare(query);
+    Span span = buildSpan(query);
+    try {
+      PreparedStatement resultSet = session.prepare(query);
+      finishSpan(span);
+      return resultSet;
+    } catch (Exception e) {
+      finishSpan(span, e);
+      throw e;
+    }
   }
 
+
   /**
    * {@inheritDoc}
    */
   @Override
   public PreparedStatement prepare(RegularStatement statement) {
-    return session.prepare(statement);
+    String query = getQuery(statement);
+    Span span = buildSpan(query);
+    try {
+      PreparedStatement resultSet = session.prepare(statement);
+      finishSpan(span);
```

Comment:
shouldn't span created (started) when prepared statement started to execute.
and span finished when prepared statement finished?
looks like here creation of statement is traced, but not execution.

Reply:
ha! you're right ;)

I will :eyes: into it